jdbi/src/main/java/org/skife/jdbi/v2/sqlobject/SqlObject.java

```diff
@@ -57,7 +58,7 @@
     {
         final SqlObject so = new SqlObject(buildHandlersFor(sqlObjectType), handle);
 
-        TypeCache<Class<?>> typeCache = new TypeCache<>(TypeCache.Sort.SOFT);
+        TypeCache<Class<?>> typeCache = new WithInlineExpunction<>(TypeCache.Sort.SOFT);
```

Comment:
`typeCache` needs to be static and shared globally, otherwise it's not really a cache :-)

Reply:
Yes, I agree. This is what I tried first ([please check these changes](https://github.com/killbill/killbill-commons/pull/103/commits/8f9b01707074a0a41764822e17339c4321a76890#diff-9a278e90e2caa8deccb8c68b2da98c3596f7b16cec425af224ecaedde26b4837R49)) and it resulted in some weird behavior ([check this build failure](https://github.com/killbill/killbill-commons/actions/runs/1562949064)).
Currently, with this change, I'm checking if these errors have anything to do with the WithInLineExpunction object type. If it works and for some reason, instance-level instantiation doesn't, then I'm thinking about checking/loading if an instance of TypeCache class type already exists in memory otherwise create a new instance.